leshan-bsserver-demo/src/main/java/org/eclipse/leshan/server/bootstrap/demo/servlet/BootstrapServlet.java

```diff
@@ -140,7 +140,13 @@ protected void doDelete(HttpServletRequest req, HttpServletResponse resp) throws
         if (bsStore.deleteConfig(endpoint)) {
             resp.setStatus(HttpServletResponse.SC_NO_CONTENT);
         } else {
-            resp.sendError(HttpServletResponse.SC_NOT_FOUND);
+            sendError(resp, HttpServletResponse.SC_NOT_FOUND, "no config for " + endpoint);
         }
     }
+
+    private void sendError(HttpServletResponse resp, int statusCode, String errorMessage) throws IOException {
+        resp.setStatus(statusCode);
+        if (errorMessage != null)
+            resp.getOutputStream().write(errorMessage.getBytes(StandardCharsets.UTF_8));
```

Comment:
I don't think we need to flush the stream : https://stackoverflow.com/questions/5043657/do-i-need-to-flush-the-servlet-outputstream